### DISCUSSION TITLE
Example of adding a new panel

### DISCUSSION THREAD

**Discussion by kadaan** [jesseduffield/lazygit#2145] [Q&A]
@jesseduffield I’d like to add at least basic worktree support.  Can you point me at a commit that adds a new panel and all necessary components?

**Comment by jesseduffield** [comment]
@kadaan I don't think we've added a panel in a while and there's been some refactorings since then. (There's more refactoring to go before it becomes an easy process). I'm currently busy on a lengthy business trip at the moment so may not have time to give you a run-down for a bit

**Reply by kadaan** [reply]
No worries. I’m working my way through it. Got the new Worktrees panel showing next to Tags

**Reply by kadaan** [reply]
@jesseduffield If I can get at least basic worktree support working in a single PR (with your guidance), it should be a nice example of how to add a panel, load new data, etc.

**Reply by kadaan** [reply]
@jesseduffield  Here is a very basic version of a new worktree panel.

![CleanShot 2022-09-01 at 11 03 26](https://user-images.githubusercontent.com/1904898/187982853-bca41904-b654-4f28-97ef-f30bf74df608.gif)

**Reply by kadaan** [reply]
@jesseduffield Created a draft PR for the work so far, so I can get some feedback. https://github.com/jesseduffield/lazygit/pull/2147
